src/app/src/main/java/com/example/bookmark/PendingRequestsActivity.java

```diff
@@ -1,18 +1,102 @@
 package com.example.bookmark;
 
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ListView;
+
+import androidx.appcompat.app.ActionBar;
 import androidx.appcompat.app.AppCompatActivity;
 
-import android.os.Bundle;
+import com.example.bookmark.adapters.BookList;
+import com.example.bookmark.models.Book;
+import com.example.bookmark.models.Owner;
+import com.example.bookmark.models.Request;
+
+import java.util.ArrayList;
+import java.util.List;
 
 /**
- * TODO: Description of class.
+ * This activity shows a user a list of books that they have pending requests
+ * for. They can select a book which takes them to the
+ * RequestedBookDetailsActivity where they can see the books details.
+ * TODO what else do we want here?
+ *
  * @author Ryan Kortbeek.
  */
 public class PendingRequestsActivity extends AppCompatActivity {
 
+    List<Book> requestedBooks = new ArrayList<>();
+    BookList requestedBooksAdapter;
+    ListView requestedBooksListView;
+
+    ActionBar pendingRequestsActionBar;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_pending_requests);
+
+        pendingRequestsActionBar = getSupportActionBar();
+        assert pendingRequestsActionBar != null;
```

Comment:
We probably shouldn't be using asserts in an Android app. This might cause the app to crash in debug or might get optimized out in a production built. We should be logging an error, throwing an exception, or both.

Reply:
Because getSupportActionBar() is not dependant on anything related to the user or their input (it just gets access to the menu so the title can be changed) I feel like its probably safe to just remove the assertion in this case